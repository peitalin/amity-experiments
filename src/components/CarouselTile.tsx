
import * as React from 'react'


interface ReactProps {
  onClick?(): void
  img: string
}

export default class CarouselTile extends React.Component<ReactProps, any> {

  componentDidMount() {
  }

  render() {
    return (
      <div className='tile__container' onClick={this.props.onClick}>
        {(
          this.props.img &&
          <div className="tile__media">
            <img className="tile__img" src={this.props.img}/>
          </div>

        )}
        <div className="tile__details">
          <div className="tile__title">
            { this.props.children }
          </div>
        </div>
      </div>
    )
  }
}
