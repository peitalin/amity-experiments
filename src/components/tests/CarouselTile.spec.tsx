

import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { mount, shallow } from 'enzyme'

import CarouselTile from '../CarouselTile'


test('<CarouselTile /> component should contain 1 .tile__container class', () => {
  const el = shallow( <CarouselTile /> )
  expect(el.find('.tile__container').length).toEqual(1)
})

test('<CarouselTile /> component should contain 1 img', () => {
  const el = shallow( <CarouselTile /> )
  expect(el.find('img').length).toEqual(1)
})

test('<CarouselTile /> component matches snapshot', () => {
  const elR = renderer.create( <CarouselTile /> )
  expect(elR.toJSON()).toMatchSnapshot()
})

const jestSpy = jest.fn()
const cTile = mount(<CarouselTile onClick={ jestSpy }/>)

test('<CarouselTile /> should have a function in its props', () => {
  expect(cTile.props().onClick).toBeDefined()
})

test('<CarouselTile /> Button should be clicked, then clicked twice', () => {
  cTile.find('.tile__container').simulate('click')
  expect(jestSpy).toHaveBeenCalledTimes(1)
  cTile.find('.tile__container').simulate('click')
  cTile.find('.tile__container').simulate('click')
  expect(jestSpy).toHaveBeenCalledTimes(3)
})


