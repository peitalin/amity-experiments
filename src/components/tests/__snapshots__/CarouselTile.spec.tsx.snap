// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarouselTile /> component matches snapshot 1`] = `
<div
  className="tile__container"
  onClick={undefined}
>
  <div
    className="tile__media"
  >
    <img
      className="tile__img"
      src="https://s3-ap-southeast-2.amazonaws.com/hayekhouses/outside/1.jpg"
    />
  </div>
  <div
    className="tile__details"
  >
    <div
      className="tile__title"
    />
  </div>
</div>
`;
